persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/AlterationMyBatisRepositoryTest.java

```diff
@@ -0,0 +1,276 @@
+
+        // FIXME: whole CNA dedicated endpoint should be removed
+        entrezGeneIds = null; // only way it works; otherwise it tries to pair up geneids with alteration types with
+        List<CopyNumberCountByGene> result = alterationMyBatisRepository.getSampleCnaCounts(
+            sampleIdToProfileId, entrezGeneIds, cnaEventTypes);
+
+        Assert.assertEquals(3, result.size());
+        AlterationCountByGene result207up = result.stream().filter(r -> r.getEntrezGeneId() == 207 && r.getAlteration() == 2).findFirst().get();
+        AlterationCountByGene result207down = result.stream().filter(r -> r.getEntrezGeneId() == 207 && r.getAlteration() == -2).findFirst().get();
+        AlterationCountByGene result208up = result.stream().filter(r -> r.getEntrezGeneId() == 208).findFirst().get();
+        Assert.assertEquals((Integer) 1, result207up.getNumberOfAlteredCases());
+        Assert.assertEquals((Integer) 1, result207down.getNumberOfAlteredCases());
+        Assert.assertEquals((Integer) 1, result208up.getNumberOfAlteredCases());
+    }
+    
+    @Test
+    public void getPatientCnaCountLegacy() throws Exception {
+
+        // FIXME: whole CNA dedicated endpoint should be removed
+        entrezGeneIds = null; // only way it works; otherwise it tries to pair up geneids with alteration types with
+        List<CopyNumberCountByGene> result = alterationMyBatisRepository.getPatientCnaCounts(
+            patientIdToProfileId, entrezGeneIds, cnaEventTypes);
+
+        // For testSql.sql there are no more samples per patient for the investigated genes.
+        // Therefore, patient level counts are the same as the sample level counts.
+        Assert.assertEquals(3, result.size());
+        AlterationCountByGene result207up = result.stream().filter(r -> r.getEntrezGeneId() == 207 && r.getAlteration() == 2).findFirst().get();
+        AlterationCountByGene result207down = result.stream().filter(r -> r.getEntrezGeneId() == 207 && r.getAlteration() == -2).findFirst().get();
+        AlterationCountByGene result208up = result.stream().filter(r -> r.getEntrezGeneId() == 208).findFirst().get();
+        Assert.assertEquals((Integer) 1, result207up.getNumberOfAlteredCases());
+        Assert.assertEquals((Integer) 1, result207down.getNumberOfAlteredCases());
+        Assert.assertEquals((Integer) 1, result208up.getNumberOfAlteredCases());
+    }
+
+    @Test
+    public void getSampleAlterationCountsReturnsZeroForMutationsAndCnaSelectorsInNone() {
+
+        mutationEventTypes = Select.none();
+        cnaEventTypes = Select.none();
+        List<AlterationCountByGene> result = alterationMyBatisRepository.getSampleAlterationCounts(
+            sampleIdToProfileId, entrezGeneIds, mutationEventTypes, cnaEventTypes, QueryElement.PASS);
+
+        Assert.assertEquals(0, result.size());
+    }
+
+    @Test
+    public void getSampleAlterationCountsReturnsAllForMutationsAndCnaSelectorsInAll() {
```

Comment:
these tests are good for covering the query argument space. If there are other combinations of arguments which might present an edge case to the mapper logic for including / excluding / selecting terms for the SQL where clause, consider expanding to other permutations.

Reply:
I have added new tests that mainly test behavior for null arguments. I also implemented the Select.java pattern for entrezGeneIds.

Reply:
:+1: